fcrepo-persistence-ocfl/src/main/java/org/fcrepo/persistence/ocfl/impl/DeleteResourcePersister.java

```diff
@@ -52,16 +57,68 @@ public void persist(final OCFLPersistentStorageSession session, final ResourceOp
         final var objectSession = session.findOrCreateSession(mapping.getOcflObjectId());
         log.debug("Deleting {} from {}", resourceId, mapping.getOcflObjectId());
         if (fedoraResourceRoot.equals(resourceId)) {
-            // We are at the root of the object.
-            objectSession.deleteObject();
+            // We are at the root of the object, so delete all the data files.
+            try {
+                objectSession.listHeadSubpaths().filter(p -> !isSidecarSubpath(p)).forEach(
+                        p -> deletePathWrapped(p, objectSession));
+            } catch (final PersistentStorageRuntimeException exc) {
+                // Rethrow the exception as a checked exception
+                throw new PersistentStorageException(exc.getCause().getMessage());
+            }
         } else {
             final var relativeSubPath = relativizeSubpath(fedoraResourceRoot, operation.getResourceId());
             final var ocflSubPath = resolveOCFLSubpath(fedoraResourceRoot, relativeSubPath);
-            final var headers = readHeaders(objectSession, ocflSubPath);
-            final var filePath = resolveExtensions(ocflSubPath,
-                    !Objects.equals(NON_RDF_SOURCE.toString(), headers.getInteractionModel())
-            );
-            objectSession.delete(filePath);
+            final var headers = (ResourceHeadersImpl) readHeaders(objectSession, ocflSubPath);
+            final boolean isRdf = !Objects.equals(NON_RDF_SOURCE.toString(), headers.getInteractionModel());
+            final var filePath = resolveExtensions(ocflSubPath, isRdf);
+            deletePath(filePath, objectSession, headers);
+            if (!isRdf) {
+                // Delete the description too.
+                final var descPath = resolveExtensions(ocflSubPath + "-description", true);
```

Comment:
Would acls need to be deleted here too? I know we aren't writing any yet. Just asking out of curiosity.

Reply:
Yes, I think that would need to be checked for. As an ACL is not required we'd need to check for its existence and delete.